### PR TITLE
[BugFix] Change the statistics table to non-static to prevent CreateAnalyzer from modifying ColumnDef

### LINKED ISSUE #9390: 【statistics】analyze full table error
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
```
CREATE TABLE `duplicate_table_with_null` ( `k1`  date, `k2`  datetime, `k3`  char(20), `k4`  varchar(20), `k5`  boolean, `k6`  tinyint, `k7`  smallint, `k8`  int, `k9`  bigint, `k10` largeint, `k11` float, `k12` double, `k13` decimal(27,9) ) ENGINE=OLAP DUPLICATE KEY(`k1`, `k2`, `k3`, `k4`, `k5`) COMMENT "OLAP" DISTRIBUTED BY HASH(`k1`, `k2`, `k3`) BUCKETS 3 PROPERTIES ( "replication_num" = "3", "storage_format" = "v2" );
```
2. INSERT INTO '....'
```
curl --location-trusted -u root: -T /home/disk4/lvchenyang/code/StarRocksTest/lib/../common/data/basic_types_data -XPUT -H label:stream_load_1658997693255 http://ip:port/api/lcy/duplicate_table_with_null/_stream_load
```
3. SELECT  '....'
```analyze full table duplicate_table_with_null_partition;```
### Expected behavior (Required)
Msg_type success
### Real behavior (Required)
Msg_type error
image

fe.log
```
2022-07-29 17:27:47,421 ERROR (statistics meta manager|36) [Daemon.run():117] daemon thread got exception. name: statistics meta manager
com.starrocks.sql.analyzer.SemanticException: Invalid default value for 'ndv'
        at com.starrocks.sql.analyzer.CreateTableAnalyzer.analyze(CreateTableAnalyzer.java:224) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.visitCreateTableStatement(Analyzer.java:82) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.visitCreateTableStatement(Analyzer.java:75) ~[starrocks-fe.jar:?]
        at com.starrocks.analysis.CreateTableStmt.accept(CreateTableStmt.java:410) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.ast.AstVisitor.visit(AstVisitor.java:112) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.analyze(Analyzer.java:77) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.Analyzer.analyze(Analyzer.java:72) ~[starrocks-fe.jar:?]
        at com.starrocks.statistic.StatisticsMetaManager.createFullStatisticsTable(StatisticsMetaManager.java:226) ~[starrocks-fe.jar:?]
        at com.starrocks.statistic.StatisticsMetaManager.createTable(StatisticsMetaManager.java:313) ~[starrocks-fe.jar:?]
        at com.starrocks.statistic.StatisticsMetaManager.refreshStatisticsTable(StatisticsMetaManager.java:332) ~[starrocks-fe.jar:?]
        at com.starrocks.statistic.StatisticsMetaManager.runAfterCatalogReady(StatisticsMetaManager.java:353) ~[starrocks-fe.jar:?]
        at com.starrocks.common.util.LeaderDaemon.runOneCycle(LeaderDaemon.java:60) ~[starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
2022-07-29 17:27:49,398 WARN (domain resolver|40) [DomainResolver.resolveWithDNS():107] unknown domain name example_domain with dns: example_domain
2022-07-29 17:27:52,190 ERROR (Thread-73|137) [ReportHandler.deleteFromMeta():683] backend [10006] invalid situation. tablet[1112333] has few replica[1], replica num setting is [1]
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
+----------------------+
| current_version()    |
+----------------------+
| MAIN-RELEASE 0f606dc |
+----------------------+
```

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9390

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 41 / 42 (97.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticUtils.java|38|39|97.44%|[191]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|3|3|100.00%|[]|

**Comment by kangkaisen** [comment]
Don't  get why need to change to non-static.

**Comment by HangyuanLiu** [comment]
> Don't get why need to change to non-static.

Because the actual value of defaultValueDef will be modified in ColumnDef analyze. When using it for the second time, it will check whether the defaultValue is assigned, which will cause the Analysis to fail and the table to fail.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 45 / 46 (97.83%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticUtils.java|42|43|97.67%|[196]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|3|3|100.00%|[]|

**Review by Seaven — APPROVED** [review]

**Review by liuyehcf — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/statistic/StatisticUtils.java [hunk 1]
```diff
@@ -121,4 +127,73 @@ public static LocalDateTime getPartitionLastUpdateTime(Partition partition) {
     public static boolean isEmptyTable(Table table) {
         return ((OlapTable) table).getPartitions().stream().noneMatch(Partition::hasData);
     }
+
+    public static List<ColumnDef> buildStatsColumnDef(String tableName) {
+        ScalarType columnNameType = ScalarType.createVarcharType(65530);
```
[the remark this answers is not in the document]
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/statistic/StatisticUtils.java:132]: Following the old code, it seems that the length here does not have much effect

**Pull request merged**